src/base/utils/string.cpp

```diff
@@ -45,114 +45,112 @@ namespace
-                // Do another round
+                else if (leftChar.isDigit() && rightChar.isDigit()) {
+                    // Both are digits, compare the numbers
+                    const auto consumeNumber = [](const QString &str, int &pos) -> int
+                    {
+                        const int start = pos;
+                        while ((pos < str.size()) && str[pos].isDigit())
+                            ++pos;
+                        return str.midRef(start, (pos - start)).toInt();
+                    };
+
+                    const int numL = consumeNumber(left, posL);
+                    const int numR = consumeNumber(right, posR);
+                    if (numL != numR)
+                        return (numL - numR);
+
+                    // String + digits do match and we haven't hit the end of both strings
+                    // then continue to consume the remainings
+                }
+                else {
+                    return (leftChar.unicode() - rightChar.unicode());
+                }
             }
-            return false;
         }
 
-    private:
         QCollator m_collator;
-        const bool m_caseSensitive;
+        const bool m_isCaseSensitive;
     };
 }
 
-bool Utils::String::naturalCompareCaseSensitive(const QString &left, const QString &right)
+int Utils::String::naturalCompare(const QString &left, const QString &right, const Qt::CaseSensitivity sensitivity)
 {
     // provide a single `NaturalCompare` instance for easy use
     // https://doc.qt.io/qt-5/threads-reentrancy.html
+    if (sensitivity == Qt::CaseSensitive) {
 #ifdef Q_OS_MAC  // workaround for Apple xcode: https://stackoverflow.com/a/29929949
-    static QThreadStorage<NaturalCompare> nCmp;
-    if (!nCmp.hasLocalData()) nCmp.setLocalData(NaturalCompare(true));
-    return (nCmp.localData())(left, right);
+        static QThreadStorage<NaturalCompare> nCmp;
+        if (!nCmp.hasLocalData())
+            nCmp.setLocalData(NaturalCompare(true));
+        return (nCmp.localData())(left, right);
 #else
-    thread_local NaturalCompare nCmp(true);
-    return nCmp(left, right);
+        thread_local NaturalCompare nCmp(true);
+        return nCmp(left, right);
 #endif
-}
+    }
 
-bool Utils::String::naturalCompareCaseInsensitive(const QString &left, const QString &right)
-{
-    // provide a single `NaturalCompare` instance for easy use
-    // https://doc.qt.io/qt-5/threads-reentrancy.html
-#ifdef Q_OS_MAC  // workaround for Apple xcode: https://stackoverflow.com/a/29929949
+#ifdef Q_OS_MAC
     static QThreadStorage<NaturalCompare> nCmp;
-    if (!nCmp.hasLocalData()) nCmp.setLocalData(NaturalCompare(false));
+    if (!nCmp.hasLocalData())
+        nCmp.setLocalData(NaturalCompare(false));
     return (nCmp.localData())(left, right);
 #else
     thread_local NaturalCompare nCmp(false);
     return nCmp(left, right);
 #endif
 }
 
+bool Utils::String::naturalLessThan(const QString &left, const QString &right)
```

Comment:
Would make it template with param of Qt::CaseSensitivity type.

Reply:
both done.